src/components/SignatureRequest.js

```diff
@@ -0,0 +1,180 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import styled from 'styled-components'
+import { Button, GU, textStyle, useTheme } from '@aragon/ui'
+import { useWallet } from '../providers/Wallet'
+import { getProviderFromUseWalletId } from '../ethereum-providers'
+import { signMessage } from '../lib/web3-utils'
+import { dayjs } from '../utils/date-utils'
+
+import signRequestSuccessIllustration from '../../src/assets/signRequestSuccess.svg'
+import signRequestFailIllustration from '../../src/assets/signRequestFail.svg'
+import signProcessing from '../../src/assets/loader.gif'
+
+const SignerRequest = React.memo(function SignerRequest({
+  actionText,
+  compactMode,
+  onActionSuccess,
+  onSignSuccess,
+  successText,
+  title,
+}) {
+  const [signingError, setSigningError] = useState(false)
+  const [signHash, setSignHash] = useState('')
+
+  const wallet = useWallet()
+
+  const provider = getProviderFromUseWalletId(wallet.activated)
+
+  const theme = useTheme()
+
+  const illustration = useMemo(() => {
+    if (signingError) {
+      return signRequestFailIllustration
+    }
+    if (signHash) {
+      return signRequestSuccessIllustration
+    }
+    return signProcessing
+  }, [signingError, signHash])
+
+  const { statusText, statusTextColor } = useMemo(() => {
+    if (signingError) {
+      return { statusText: 'Signature failed', statusTextColor: theme.negative }
+    }
+    if (signHash) {
+      return {
+        statusText: 'Signature confirmed',
+        statusTextColor: theme.positive,
+      }
+    }
+    return {
+      statusText: 'Waiting for signature…',
+      statusTextColor: theme.surfaceContentSecondary,
+    }
+  }, [signingError, signHash, theme])
+
+  const infoText = useMemo(() => {
+    if (signingError) {
+      return 'An error has occurred when signing the message.'
+    }
+    if (signHash) {
+      return successText
+    }
+    return `Open ${provider.name} to complete the
+    signature request. Signing this message will prove ownership of your
+    account and ${actionText}`
+  }, [actionText, provider, signingError, signHash, successText])
+
+  const requestSignature = useCallback(async () => {
+    if (!wallet) {
+      return
+    }
+    const now = dayjs().toString()
+    const { signHash, error } = await signMessage(wallet, now)
+
+    if (error) {
+      setSigningError(true)
+      return
+    }
+
+    if (typeof onSignSuccess === 'function') {
```

Comment:
`onSignSuccess()` is really meant to be a required prop; this component doesn't work at all if it's not provided (why sign a message if there's nothing to give it back to?).

Although we're not using prop types, it still seems reasonable for us to drop the typecheck here to make it more explicit that we expect this to be passed as a prop.